File: python/02-FundamentalsII/01-funII.py
#1 - Sigma
def sigma(num):
    sum = 0
    for i in range(num+1):
        sum += i
    return sum
#print(sigma(5))

#2 - Factorial
def factorial(num):
    fact = 1
    for i in range(1,num+1):
        fact *= i
    return fact
#print(factorial(5))

#3.1 - Star Art
def drawLeftStars(num):
    print("*"*num)
#drawLeftStars(25)

#3.2 - Star Art
def drawRightStars(num):
    if num > 74:
        return "{} Excedes character limit.".format(num)
    if num < 1:
        return "Please input a positive integer value of characters."
    return "|"+" "*(75-num)+"*"*num
#print(drawRightStars(285))

#3.3 - Star Art
def drawCenteredStars(num):
    if num > 73:
        print(num)
        return "Input excedes character limit."
    if num < 1:
        return "Please input a positive integer value of characters."
    outside = (75-num)/2
    return "|"+" "*outside+"*"*num+" "*outside+"|"
#print(drawCenteredStars(30))

#3.4 - Star Art
def starWars():
    xWing = ">o<"
    tieF = "|=|"
    yWing = ">-=D"
    print(xWing+"     "+tieF+"     "+yWing)
#starWars()

#4.1 - Character Art
def drawLeftChars(num,char):
    print(char*num)
#drawLeftChars(10, "X")

#4.2 - Character Art
def drawRightChars(num,char):
    if num > 74:
        return "{} Excedes character limit.".format(num)
    if num < 1:
        return "Please input a positive integer value of characters."
    return "|"+" "*(75-num)+(char*num)
#print(drawRightChars(20, "X"))

#4.3 - Character Art
def drawCenteredChars(num,char):
    if num > 73:
        return "{} Excedes character limit.".format(num)
    if num < 1:
        return "Please input a positive integer value of characters."
    boundary = (75-num)/2
    return "|"+" "*boundary+"*"*num+" "*boundary+"|"
#print(drawCenteredChars(45, "X"))

File: javaScript/02-FundamentalsII/01-funII.js
// 1 - Sigma
function sigma(num){
    sum = 0;
    for(i = 1; i <= num; i++){
        sum += i;
    }
    return sum;
}
// console.log(sigma(5));

// 2 - Factorial
function factorial(num){
    fact = 1;
    for(i = 1; i <= num; i++){
        fact *= i;
    }
    return fact;
}
// console.log(factorial(5))

// 3 - Star Art
// 3.1
function drawLeftStars(num){
    output = "|";
    if(num > 74){
        return num+" Excedes character limit.";
    }
    if(num < 1){
        return "Please input a positive integer value of characters.";
    }
    for(i = 0; i <= num; i++){
        output += "*";
    }
    for(i = 0; i <= (75-num); i++){
        output += " ";
    }
    output += "|";
    return output;
}
console.log(drawLeftStars(60));

// 3.2
function drawRightStars(num){
    output = "|";
    if(num > 74){
        return num+" Excedes character limit.";
    }
    if(num < 1){
        return "Please input a positive integer value of characters.";
    }
    for(i = 0; i <= (75-num); i++){
        output += " ";
    }
    for(i = 0; i <= num; i++){
        output += "*";
    }
    output += "|";
    return output;
}
console.log(drawRightStars(60));

// 3.3
function drawCenteredStars(num){
    output = "|";
    boundary = (75-num)/2;
    if (num > 73){
        return num+" excedes character limit.";
    }
    if (num < 1){
        return "Please input a positive integer value of characters.";
    }
    for(i = 0; i <= boundary; i++){
        output += " ";
    }
    for(i = 0; i <= num; i++){
        output += "*";
    }
    for(i = 0; i <= boundary; i++){
        output += " ";
    }
    output += "|";
    return output;
}
console.log(drawCenteredStars(60));

